confluent_kafka/src/Consumer.c

```diff
@@ -391,6 +391,55 @@ static PyObject *Consumer_position (Handle *self, PyObject *args,
 	return plist;
 }
 
+static PyObject *Consumer_pause(Handle *self, PyObject *args,
+                    PyObject *kwargs) {
+
+    PyObject *plist;
+	rd_kafka_topic_partition_list_t *c_parts;
+    rd_kafka_resp_err_t err;
+    static char *kws[] = {"partitions", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O", kws, &plist))
+        return NULL;
+
+    if (!(c_parts = py_to_c_parts(plist)))
+        return NULL;
+
+    err = rd_kafka_pause_partitions(self->rk, c_parts);
+    if (err) {
+        rd_kafka_topic_partition_list_destroy(c_parts);
+        cfl_PyErr_Format(err,
+                "Failed to pause partitions: %s",
+                rd_kafka_err2str(err));
+        return NULL;
+    }
+	Py_RETURN_NONE;
+}
+
+static PyObject *Consumer_resume (Handle *self, PyObject *args,
+                    PyObject *kwargs) {
+
+    PyObject *plist;
+	rd_kafka_topic_partition_list_t *c_parts;
+    rd_kafka_resp_err_t err;
+    static char *kws[] = {"partitions", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O", kws, &plist))
+        return NULL;
+
+    if (!(c_parts = py_to_c_parts(plist)))
+        return NULL;
+
+    err = rd_kafka_resume_partitions(self->rk, c_parts);
+    if (err) {
```

Comment:
c_parts will leak in the success case.
Move list_destroy(c_parts) up here so it won't leak when !err